src/compile/common.ts

```diff
@@ -114,9 +139,6 @@ export function formatSignalRef(fieldDef: FieldDef<string>, specifiedFormat: str
 export function numberFormat(fieldDef: FieldDef<string>, specifiedFormat: string, config: Config, channel: Channel) {
   // Specified format in axis/legend has higher precedence than fieldDef.format
   const format = specifiedFormat;
-  if (fieldDef.type === QUANTITATIVE) {
```

Comment:
@kanitw  I need advice about what I should do here? When should I return a number format apart from when fieldDef.type is quantitative? I think my visualizations are fine but it's still bad style and it might be breaking something else somewhere. What if the type is O, and the user wants to get integers?